### ISSUE TITLE
Cannot compile on OSX High Sierra - auto_ptr not in namespace std 

### ISSUE THREAD

**Issue by yamamushi** [Open-Transactions/Moneychanger#308]
`/Users/yamamushi/programming/projects/Moneychanger/src/core/handlers/contacthandler.cpp:1362:26: error: no member named 'auto_ptr' in
      namespace 'std'
                    std::auto_ptr<DBHandler::PreparedQuery> qu;
                    ~~~~~^
/Users/yamamushi/programming/projects/Moneychanger/src/core/handlers/contacthandler.cpp:1362:59: error: expected '(' for function-style
      cast or type construction
                    std::auto_ptr<DBHandler::PreparedQuery> qu;`



This is also triggered in xmlrpc-c

`/usr/local/include/xmlrpc-c/girmem.hpp:45:10: error: no template named 'auto_ptr' in namespace 'std'
    std::auto_ptr<Impl> const implP;
    ~~~~~^
/usr/local/include/xmlrpc-c/girmem.hpp:45:25: error: expected member name or ';' after declaration specifiers
    std::auto_ptr<Impl> const implP;
    ~~~~~~~~~~~~~       ^`

And 

`In file included from /usr/local/include/xmlrpc-c/client.hpp:10:
/usr/local/include/xmlrpc-c/girmem.hpp:45:10: error: no template named 'auto_ptr' in namespace 'std'
    std::auto_ptr<Impl> const implP;
    ~~~~~^
/usr/local/include/xmlrpc-c/girmem.hpp:45:25: error: expected member name or ';' after declaration specifiers
    std::auto_ptr<Impl> const implP;`


I'm still looking into how to resolve this.

**Comment by yamamushi** [comment]
Resolved by a find and replace for auto_ptr -> unique_ptr in the affected files.

Arguably, that's not the preferred long-term solution, but it got me to where I needed to be. 

I also had to downgrade from Qt 5.11 to 5.4, and remove the homebrew qt bundle in favor of the one downloaded directly from qt.io (I also added `~/Qt/5.4/clang_64/bin` to my path in .bashrc). This was necessary because Qt 5.11 from homebrew was throwing errors about a missing widget type. 

Still haven't gotten it to successfully build but I am still working on it.
